skiko/src/iosMain/kotlin/org/jetbrains/skiko/redrawer/MetalRedrawer.ios.kt

```diff
@@ -33,12 +36,55 @@ internal class MetalRedrawer(
     private val queue = device.newCommandQueue() ?: throw IllegalStateException("Couldn't create Metal command queue")
     private var currentDrawable: CAMetalDrawableProtocol? = null
     private val metalLayer = MetalLayer()
+
+    private var drawSchedulingState = DrawSchedulingState.AVAILABLE_ON_NEXT_FRAME
```

Comment:
Why initial value is not `AVAILABLE_ON_CURRENT_FRAME`?

Reply:
Because precise vsync timing is not accessible without CADisplayLink callbacks. Voluntarily dispatching a frame outside of its scheduling loop can cause frame drifting in worst cases.

Reply:
Please write it as a comment to this line

Reply:
Done